### DISCUSSION TITLE
UI-Studio Component for Flow that's mentioned on your page here: https://docs.totaljs.com/uibuilder/y21k001da41d/#y3qb001da41d

### DISCUSSION THREAD

**Discussion by vtali** [totaljs/flow#120] [Q&A] [answered]
Hi I was wondering, what I can do to enable my Form that I built in UI-Studio to be able to communicate with my workflow I created in Flow? 

I don't see in UI Studio where my Form can POST to Flow endpoint?

So then I saw that this component can be possibly linked with what's mentioned here: 
https://docs.totaljs.com/uibuilder/y21k001da41d/#y3qb001da41d

**Comment by petersirka** [comment, marked as answer]
H @vtali.
This functionality is part of [Total.js Enterprise Service](https://www.totaljs.com/enterprise/). BTW: we have added similar functionality to the Total.js CMS with two new CMS plugins (part of Total.js Enterprise): Flow and Forms. So, you can easily make a website, create forms, and then create your own workflows for data processing in Flow.

**Reply by vtali** [reply]
Thank you, any recommendation for my POC on how I can submit my UI-Studio form I built to an api? 
I'm thinking I can then let that be the `trigger` that processes my flow.

I appreciate the already provided solution of enterprise service, but am unable to make that commitment just yet.

**Reply by petersirka** [reply]
It's not hard. You can communicate with the designed app through the `input` method and `output` event. The important component is `form`. When the user clicks on the submit button, you can capture data in the `app.on('output', function(meta) {` event.

- Documentation  [`app.on('output', function(meta) {`](https://docs.totaljs.com/uibuilder/Ihstvu1c761f/#IhsyuF1cX61f)

The same functionality uses our j-UIStudio component for communicating with the Flow. With the `app.input()` method, you can send data to the `form`, and with the help of `app.on('output', ...)` you can capture data from the form.

**Reply by vtali** [reply]
Thank you for that
